manifests/init.pp

```diff
@@ -122,48 +122,29 @@
 # @author https://github.com/elastic/puppet-logstash/graphs/contributors
 #
 class logstash(
-  $ensure            = 'present',
-  $status            = 'enabled',
-  $restart_on_change = true,
-  $auto_upgrade       = false,
-  $version           = undef,
-  $package_url       = undef,
-  $package_name      = 'logstash',
-  $download_timeout  = 600,
-  $logstash_user     = 'logstash',
-  $logstash_group    = 'logstash',
-  $config_dir         = '/etc/logstash',
-  $purge_config      = true,
-  $service_provider  = undef,
-  $settings          = {},
-  $startup_options   = {},
-  $jvm_options       = [],
-  $manage_repo       = true,
-  $repo_version      = '5.x',
-)
+  Enum['present','absent'] $ensure                         = 'present',
```

Comment:
Maybe you want to take a look at the Vox Pupuli guide I wrote:
https://voxpupuli.org/blog/2016/12/22/putting-down-puppet-3/
https://voxpupuli.org/blog/2017/01/11/migrating-to-puppet4/